test-utils/src/dev/db.rs

```diff
@@ -377,8 +377,27 @@ impl CockroachStarter {
                                 )
                             })
                         }
-
-                        _ => Err(poll::CondCheckError::NotYet),
+                        Ok(_) => {
+                            // The file hasn't been fully written yet.
+                            Err(poll::CondCheckError::NotYet)
+                        }
+                        Err(error) => {
+                            let maybe_errno = error.raw_os_error();
+                            if maybe_errno.is_some()
+                                && maybe_errno.unwrap() == libc::ENOENT
```

Comment:
If you don't like the `is_some`/`unwrap` here, I think it'd be equivalent to check `error.kind == io::ErrorKind::NotFound`: https://github.com/rust-lang/rust/blob/b70baa4f922a1809d79caeaeb902800c3be283b9/library/std/src/sys/unix/mod.rs#L252

Reply:
Thanks.  I think that will make this much cleaner.  Fixing in 546b63c9.